SpatialGDK/Documentation/content/tutorials/multiserver-shooter/tutorial-multiserver-localtest.md

```diff
@@ -0,0 +1,112 @@
+5. Using the Inspector to track the location of your two players, notice that if you position them in the same area of authority then their shots damage each other, but if they are on different servers, they can’t damage each other. Let’s fix that.
+
+<br/>
+### Enable cross server RPCs
+
+To damage a player on a different server, the actor shooting the bullet must send a cross-server RPC to the actor getting hit by the bullet. You will implement this by overriding the [TakeDamage (Unreal documentation)](https://api.unrealengine.com/INT/API/Runtime/Engine/GameFramework/APawn/TakeDamage/index.html) function in the TPSCharcter class.
+
+1. In your IDE, open `UnrealGDKThirdPersonShooter\Game\Source\ThirdPersonShooter\Characters\TPSCharacter.h`.
+2. On line 74, add this snippet:
+
+    ```
+    UFUNCTION(CrossServer, Reliable)
+    void TakeGunDamageCrossServer(float Damage, const struct FDamageEvent& DamageEvent, AController* EventInstigator, AActor* DamageCauser);
+    ```
+
+    This snippet creates a new `UFUNCTION` marked with the function tags [CrossServer]({{urlRoot}}/content/cross-server-rpcs) and [Reliable (Unreal documentation)](https://wiki.unrealengine.com/Replication#Reliable_vs_Unreliable_Function_Call_Replication). The CrossServer tag forces this function to be executed as a cross-server RPC.
+
+3. In your IDE, open `UnrealGDKThirdPersonShooter\Game\Source\ThirdPersonShooter\Characters\TPSCharacter.cpp`.
+4. Replace the `TakeDamage` function (lines 514-548) with this snippet:
+
+```
+float ATPSCharacter::TakeDamage(float Damage, const FDamageEvent& DamageEvent, AController* EventInstigator, AActor* DamageCauser)
+{
+	TakeGunDamageCrossServer(Damage, DamageEvent, EventInstigator, DamageCauser);
+
+	return Damage;
+}
+
+void ATPSCharacter::TakeGunDamageCrossServer_Implementation(float Damage, const FDamageEvent& DamageEvent, AController* EventInstigator, AActor* DamageCauser)
+{
+	if (!HasAuthority())
+	{
+		return;
+	}
+
+	const ATPSCharacter* Killer = nullptr;
+
+	// Ignore friendly fire
+	const AInstantWeapon* DamageSourceWeapon = Cast<AInstantWeapon>(DamageCauser);
+	if (DamageSourceWeapon != nullptr)
+	{
+		const ATPSCharacter* DamageDealer = Cast<ATPSCharacter>(DamageSourceWeapon->GetWeilder());
+		if (DamageDealer != nullptr)
+		{
+			if (Team != ETPSTeam::Team_None    // "Team_None" is not actually a team, and "teamless" should be able to damage one another
+				&& DamageDealer->GetTeam() == Team)
+			{
+				return;
+			}
+			Killer = DamageDealer;
+		}
+	}
+
+	int32 DamageDealt = FMath::Min(static_cast<int32>(Damage), CurrentHealth);
+	CurrentHealth -= DamageDealt;
+
+	if (CurrentHealth <= 0)
+	{
+		Die(Killer);
+	}
+}
+```
+
+This snippet implements the functionality that was previously contained within `TakeDamage` as a cross-server RPC called `TakeGunDamageCrossServer`.
+
+Because you’ve changed code in a function you now need build **ThirdPersonShooter.sln**, generate schema and a new snapshot. To do this:
```

Comment:
```suggestion
Because you’ve changed code in a function, you now need build **ThirdPersonShooter.sln**, generate schema, and generate a new snapshot. To do this:
```